plugins/modules/foreman_setting.py

```diff
@@ -87,30 +81,34 @@ class ForemanSettingModule(ForemanAnsibleModule):
 
 def main():
     module = ForemanSettingModule(
-        argument_spec=dict(
+        foreman_spec=dict(
             name=dict(required=True),
             value=dict(type='raw'),
         ),
     )
 
-    module_params = module.clean_params()
+    # TODO Maybe refactor this into a EntityMixin
+    module.foreman_spec.update(_foreman_spec_helper(dict(
```

Comment:
calling a private helper here feels weird. and the code is hard to read. the normal magic can't help here as it's not an entity module?

Reply:
Well, it could be an Entity module if that would not imply state.

There could be a lot more separation of concerns...